dhall/src/Dhall/TypeCheck.hs

```diff
@@ -4252,21 +4252,23 @@ data TypeError s a = TypeError
     }
 
 instance (Eq a, Pretty s, Pretty a) => Show (TypeError s a) where
-    show = Pretty.renderString . Dhall.Pretty.layout . Pretty.pretty
+    show = Pretty.renderString . Dhall.Pretty.layout . prettyTypeError
```

Comment:
Sorry, this is actually unrelated – it just allows us to skip `unAnnotate` which can be expensive for large `Doc`s (which isn't the case here).